lib/hardwarekey/agent.go

```diff
@@ -101,11 +102,15 @@ func NewAgentServer(ctx context.Context, s hardwarekey.Service, keyAgentDir stri
 }
 
 func newAgentListener(ctx context.Context, keyAgentDir string) (net.Listener, error) {
+	// On windows, listening on an already existing unix socket results in a different,
+	// Windows specific [syscall.Bind] error not defined in the [syscall] library.
+	const windowsBindErrMessage = "bind: Only one usage of each socket address (protocol/network address/port) is normally permitted"
```

Comment:
I worry that error message strings are not stable. At the very least, I would remove the `bind: ` prefix as that looks like it comes from error wrapping and not part of the original Windows error message. An alternative is to use the error code if that is available: `WSAEADDRINUSE` or `10048`. The symbolic value can be found in package `golang.org/x/sys/windows` (https://pkg.go.dev/golang.org/x/sys@v0.32.0/windows), but that may be too messy as the error type from Windows syscalls is only defined for Windows builds.

Just a thought; not blocking approval for this.

Reply:
Yeah you're right, I was considering adding a build constraint but couldn't find the actual location of the Windows error. Thanks for uncovering that for me. e7cddc3